graylog2-web-interface/src/components/common/ConfirmDialog.tsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Modal, Button } from 'components/graylog';
+
+/**
+ * Component that displays a confirmation dialog box that the user can
+ * cancel or confirm.
+ */
+const ConfirmDialog = ({
+  show,
+  onDialogHide,
+  onDialogOpen,
+  onDialogClose,
+  title,
+  children,
+  onCancel,
+  onConfirm,
+  btnCancelDisabled,
+  btnConfirmDisabled,
+  btnCancelText,
+  btnConfirmText,
+}) => {
+  return (
+    <Modal show={show} onHide={onDialogHide} onEnter={onDialogOpen} onExit={onDialogClose}>
+      <Modal.Header closeButton>
+        <Modal.Title>{title}</Modal.Title>
+      </Modal.Header>
+
+      <Modal.Body>
+        {children}
+      </Modal.Body>
+
+      <Modal.Footer>
+        <Button type="button" onClick={onCancel} disabled={btnCancelDisabled}>{btnCancelText}</Button>
```

Comment:
What's the difference between `onDialogHide`, `onDialogClose`, and `onCancel`? Shouldn't all cancel the confirm dialog?

Reply:
No, because if you cancel the dialog, then it will just close the modal.

Reply:
I also forget a few words there 😄 👌